numba/cuda/tests/cudapy/test_atomics.py

```diff
@@ -261,15 +289,114 @@ def test_atomic_add_double_global_3(self):
 
         np.testing.assert_equal(ary, orig + 1)
 
+    def test_atomic_max_int32(self):
```

Comment:
There's a lot of repetition amongst the tests for various dtypes, perhaps the repetition can be factored out?